contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1290,62 +1203,12 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
     }
 
     /**
-     * @dev returns the protection level based on the timestamp and protection delays
+     * @dev deducts the IL amount from the given position value
      */
-    function _protectionLevel(uint256 addTimestamp, uint256 removeTimestamp) internal view returns (Fraction memory) {
-        uint256 timeElapsed = removeTimestamp.sub(addTimestamp);
-        uint256 minProtectionDelay = _settings.minProtectionDelay();
-        uint256 maxProtectionDelay = _settings.maxProtectionDelay();
-        if (timeElapsed < minProtectionDelay) {
-            return Fraction({ n: 0, d: 1 });
-        }
-
-        if (timeElapsed >= maxProtectionDelay) {
-            return Fraction({ n: 1, d: 1 });
-        }
-
-        return Fraction({ n: timeElapsed, d: maxProtectionDelay });
-    }
-
-    /**
-     * @dev returns the compensation amount based on the impermanent loss and the protection level
-     */
-    function _compensationAmount(
-        uint256 amount,
-        uint256 total,
-        Fraction memory loss,
-        Fraction memory level
-    ) internal pure returns (uint256) {
-        uint256 levelN = level.n.mul(amount);
-        uint256 levelD = level.d;
-        uint256 maxVal = Math.max(Math.max(levelN, levelD), total);
+    function _deductIL(uint256 value, Fraction memory loss) internal pure returns (uint256) {
+        uint256 maxVal = Math.max(1, value);
```

Comment:
I'm pretty sure that `value` is always larger than 0, making this redundant (i.e., we can just use `value` instead of `maxVal`).

Reply:
I believe it's your code so I assume you had a reason for it, might as well keep it and not get surprised later :)